src/main/java/core/basesyntax/ReportGenerator.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ReportGenerator {
+
+    private final Map<Operation, OperationHandler> operationHandlers = new HashMap<>();
+    private final FruitStock fruitStock = new FruitStock(); // Створюємо екземпляр FruitStock
+
+    public ReportGenerator() {
+        operationHandlers.put(Operation.SUPPLY, new SupplyHandler(fruitStock));
+        operationHandlers.put(Operation.PURCHASE, new PurchaseHandler(fruitStock));
+        operationHandlers.put(Operation.RETURN, new ReturnHandler(fruitStock));
+    }
+
+    public String generateReport(List<FruitTransaction> transactions) {
+        Map<String, Integer> report = new HashMap<>();
```

Comment:
```suggestion
    public String generateReport() {
        Map<String, Integer> report = fruitStock.stock;
```
use map from the fruitStock